examples/wasm/iwasmt.c

```diff
@@ -0,0 +1,157 @@
+        8 * 1024, error_buf, sizeof(error_buf)))) {
+        puts(error_buf);
+    }else{
+        app_instance_main(wasm_module_inst);
+        /* destroy the module instance */
+        wasm_runtime_deinstantiate(wasm_module_inst);
+    }
+    return NULL;
+}
+
+void *
+iwasm_main(void *arg1)
+{
+    (void) arg1; /*unused*/
+    uint8_t *wasm_file_buf = NULL;
+    unsigned wasm_file_buf_size = 0;
+    wasm_module_t wasm_module = NULL;
+    char error_buf[128];
+
+    RuntimeInitArgs init_args;
+
+//chose allocator non defaults to system allocator
+#define FUNC_ALLOC
+//#define POOL_ALLOC
+
+    memset(&init_args, 0, sizeof(RuntimeInitArgs));
+#ifdef POOL_ALLOC
+    static char global_heap_buf[256 * 1024] = { 0 };//(256 kB)
+
+    init_args.mem_alloc_type = Alloc_With_Pool;
+    init_args.mem_alloc_option.pool.heap_buf = global_heap_buf;
+    init_args.mem_alloc_option.pool.heap_size = sizeof(global_heap_buf);
+#elif defined(FUNC_ALLOC)
+    init_args.mem_alloc_type = Alloc_With_Allocator;
+/*deactivate Wpedantic for some lines*/
+#pragma GCC diagnostic ignored "-Wpedantic"
+    init_args.mem_alloc_option.allocator.malloc_func = malloc;
+    init_args.mem_alloc_option.allocator.realloc_func = realloc;
+    init_args.mem_alloc_option.allocator.free_func = free;
+#pragma GCC diagnostic pop
+#else
+    init_args.mem_alloc_type = Alloc_With_System_Allocator;
+#endif
+
+    /* initialize runtime environment */
+    if (!wasm_runtime_full_init(&init_args)) {
+        puts("Init runtime environment failed.");
+        return NULL;
+    }
+
+    /* we need the byte codes be writable while loading it adds size information to
+    /* stack POPs */
+    //static uint8_t wasm_test_file[] = main_wasm;
+    wasm_file_buf = malloc(main_wasm_len);
+    memcpy(wasm_file_buf, main_wasm, main_wasm_len);
+    /* if architecture has const in writable mem
+    /* wasm_file_buf = (uint8_t *) main_wasm; */
+    wasm_file_buf_size = main_wasm_len;
+
+    /* load WASM module */
+    if (!(wasm_module = wasm_runtime_load(wasm_file_buf, wasm_file_buf_size,
+                                          error_buf, sizeof(error_buf)))) {
+        puts(error_buf);
+
+    }else{
+        iwasm_t(wasm_module);
+        wasm_runtime_unload(wasm_module);
+    }
+
+    wasm_runtime_destroy();
+    return NULL;
+}
+
+bool
+iwasm_init(void)
+{
+    struct{
+        char *  stack;
+        int  stacksize;
+        uint8_t  priority;
+        int  flags;
+        thread_task_func_t task_func;
+        void *  arg;
+        const char * name;
+    } b = {
+        .stacksize = DEFAULT_THREAD_STACKSIZE,
+        .priority = 8,
+        .flags = 0,
+        .task_func = iwasm_main,
+        .arg = NULL,
+        .name = "simple_wamr"
+    };
+
+    b.stack=malloc(b.stacksize);
+    kernel_pid_t tpid = thread_create (b.stack, b.stacksize, b.priority, b.flags, b.task_func, b.arg, b.name);
+
+    return tpid != 0 ? true : false;;
+}
+
+#define telltruth(X) ((X) ? "true" : "false")
```

Comment:
Since this is used exactly once, why the effort of a macro?

Reply:
cause it was used more than once while develpoling